### PR TITLE
ICONS-KEY and CLAUSE-KEY in arg. drop rules

### LINKED ISSUE #485: Fix ICONS in clausalmods
https://delphinqa.ling.washington.edu/t/what-should-the-icons-look-like-in-a-clausalmod-sentence/544

### PR DESCRIPTION
**Pull request by olzama**
https://delphinqa.ling.washington.edu/t/what-should-the-icons-look-like-in-a-clausalmod-sentence/544/13

The subject-drop and object-drop rules were not linking the ICONS-KEY and the CLAUSE-KEY to the mother node. Subsequently ICONS were not properly formed.

### CODE REVIEW

File: tests/regression/home/gold/valchg-lkt/result [hunk 1]
```diff
@@ -11,6 +11,5 @@
...
-26@0@-1@-1@-1@-1@-1@-1@-1@-1@(28 decl-head-opt-subj 0.000000 0 1 (27 basic-head-opt-comp 0.000000 0 1 (26 caus-itr-agt-pron-1sg-subj-prefix 0.000000 0 1 (25 caus-suffix 0.000000 0 1 (24 pat-subj-2sg-pron-prefix 0.000000 0 1 (2 ičháGA 0.000000 0 1 ("1SgAgt-2SgPat-ičháGA-CAUS")))))))@@("S" ("VP" ("V" ("V" ("VP" ("VP" ("1SgAgt-2SgPat-ičháGA-CAUS")))))))@[ LTOP: h0 INDEX: e2 [ e SF: prop-or-ques E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] RELS: < [ "_ičháǧA_v_rel"<-1:-1> LBL: h4 ARG0: e5 [ e SF: iforce E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] ARG1: x6 [ x SPECI: bool COG-ST: cog-st PNG.PER: 2nd PNG.NUM: singular ] ]  [ "_caus_v_rel"<-1:-1> LBL: h1 ARG0: e2 ARG1: x3 [ x SPECI: bool COG-ST: in-foc PNG.PER: 1st PNG.NUM: singular ] ARG2: x6 ARG3: h7 ] > HCONS: < h0 qeq h1 h7 qeq h4 > ICONS: < e8 [ e SF: iforce E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] non-focus x3 e9 [ e SF: iforce E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] non-focus x6 > ]@((:ascore . 0.0) (:probability . 1.0))
```
Comment (dantiston) [inline tests/regression/home/gold/valchg-lkt/result:14]: Did we lose this parse?

File: tests/regression/home/gold/clausalmods-lavukaleve/result [hunk 2]
```diff
@@ -1,13 +1,5 @@
...
...0 1 3 (6 lai 0.000000 1 2 ("lai")) (8 ga_2 0.000000 2 3 ("ga"))) (138 agr-subj-m-suffix 0.000000 3 4 (9 oeo 0.000000 3 4 ("oeom"))))))@@("S" ("PP" ("S" ("VP" ("V" ("V" ("V" ("alevalorige"))))))) ("S" ("VP" ("NP" ("N" ("lai")) ("D" ("ga"))) ("V" ("V" ("oeom"))))))@[ LTOP: h0 INDEX: e2 [ e SF: prop-or-ques E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] RELS: < [ "_anterior_subord_rel"<-1:-1> LBL: h4 ARG0: i5 ARG1: h6 ARG2: h7 ]  [ "_wake.up_v_rel"<-1:-1> LBL: h8 ARG0: e9 [ e SF: prop-or-ques E.TENSE: tense E.ASPECT: aspect E.MOOD: mood ] ARG1: x10 [ x SPECI: bool COG-ST: in-foc PNG.PERNUM: pernum PNG.GEND: gender ] ARG2: x11 [ x SPECI: bool COG-ST: cog-st PNG.PERNUM: 3singular PNG.GEND: masculine ] ]  [ "_rain_n_rel"<-1:-1> LBL: h12 ARG0: x13 [ x SPECI: bool COG-ST: cog-st PNG.PERNUM: 3singular PNG.GEND: neuter ] ]  [ "_def_q_rel"<-1:-1> LBL: h14 ARG0: x13 RSTR: h15 BODY: h16 ]  [ "_go.all.around_v_rel"<-1:-1> LBL: h17 ARG0: e2 ARG1: x18 [ x SPECI: bool COG-ST: in-foc PNG.PERNUM: 3singular PNG.GEND: masculine ] ARG2: x13 ] > HCONS: < h0 qeq h1 h6 qeq h17 h7 qeq h8 h15 qeq h12 > ICONS: < e9 non-focus x10 e9 non-focus x11 e2 non-focus x18 > ]@((:ascore . 0.0) (:probability . 0.333333))
```
Comment (dantiston) [inline tests/regression/home/gold/clausalmods-lavukaleve/result:3]: Did we lose these parses?

**Review by dantiston — COMMENTED** [review]
Besides the possibly missing parses, looks good to me! Thanks!

**Comment by olzama** [comment]
Oh, I did not notice parses were missing (somehow). Thanks. I will look now.

**Comment by olzama** [comment]
Hm. We should not have lost any parses. Looking at the valchg-lkt test:

  Current profile: /Users/olzama/Research/matrix-git/tests/regression/home/current/valchg-lkt
  Gold profile: /Users/olzama/Research/matrix-git/tests/regression/home/gold/valchg-lkt
  1                                        <0,1,0>
  3                                        <0,1,0>
  5                                        <0,1,0>
  8                                        <0,1,0>
  10                                       <0,1,0>
  12                                       <0,1,0>
  14                                       <0,1,0>
  16                                       <0,1,0>
  18                                       <0,1,0>
  19                                       <0,1,0>
  21                                       <0,1,0>
  23                                       <0,1,0>
  25                                       <0,1,0>
  26                                       <1,0,1>
  28                                       <0,1,0>
  29                                       <0,1,0>
Result: FAIL

Screen Shot 2020-07-30 at 5 15 57 PM


The parse for number 26 is there, it is just different (and I updated it because it was better). Could it be that it is something about interpreting the diffs?..

**Comment by olzama** [comment]
It is odd... I can see that indeed, it appears as if parse number 26 was removed from the result file! Although when I look inside the file locally, I can see it (and I checked that this is the right revision):

Screen Shot 2020-07-30 at 5 30 21 PM


@goodmami do you have an idea what might be going on?? some git magic? :)

**Comment by olzama** [comment]
It's gotta be that I somehow a different version locally, right?.. I am very confused :). I only have one branch named like this...

**Comment by olzama** [comment]
Hmm. The screenshot above (from the result file, with item 26 present) is from my *current*. In the gold, item 26 is indeed missing. But why?.. And how is the test passing, then? 

(pydelphin-1.4.0) (base) MurkinPro:matrix-git olzama$ ./rtest.py valchg-lkt
valchg-lkt                                                                                                        PASS

**Comment by olzama** [comment]
Oh, perhaps, if an item is missing from the *result* file, then the test does not care whether or not it is present in the current. But why did it go missing, I don't understand.

**Comment by olzama** [comment]
OK, this is thoroughly confusing but I will try to figure it out tomorrow. There seems to be some version confusion or maybe there were some local changes somewhere lingering... I don't know. I think I can now at least reproduce this locally (not getting e.g. item 26 from valchg-lkt).

**Comment by olzama** [comment]
OK I think I have a theory of what happened. Everyone please don't yet spend your time trying to help :). I will probably be able to clear this up soon, or at least to provide a partial fix (so that we still have some loose ICONS). But at least no lost parses :). This was a combination of local changes in *current* from two separate regresstion test subset(?) runs, I think. The change in the subject drop rule has the desired result, but the change in the object drop is not good, and somehow I did not notice even though I was sure I ran all the tests once again separately after that second change. Idk. Anyway. TJ thank you for your attention to detail!

**Comment by goodmami** [comment]
I checked out this branch and for me both the `readings` in the `parse` relation is 0 and the result for item 26 is missing from the `result` relation for both gold and current. It looks like this version of the customization system is losing the parses (maybe from the stricter constraints on ICONS), and then you updated the gold profiles without inspecting them closely.

> Oh, perhaps, if an item is missing from the result file, then the test does not care whether or not it is present in the current.

This is not (or should not be) true. In this case the <gold, shared, test> triple should show something like <0,0,1>, and it would fail.

**Comment by olzama** [comment]
I've created a huge mess now locally with all the versions of these branches. I think I am going to scratch this altogether and start again from the trunk.

**Pull request closed without merging**